### DISCUSSION TITLE
javaLTSVersions not applying to docker tags from private registry

### DISCUSSION THREAD

**Discussion by cody-reibsome** [renovatebot/renovate#32474] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab - 39.9.3

### Please tell us more about your question or problem

I have various references to eclipse-temurin docker images in my codebase, but only the ones that do not include a private registry prefix seem to respect the javaLTSVersions workaround.

i.e.

sample .gitlab-ci.yml snippet:
```
default:
  image: private.registry.io/eclipse-temurin:21-alpine@sha256:c63d8669d87e16bcee66c0379d1deedf844152da449ad48f2c8bd73a3705d36b
```

Renovate will suggest an upgrade from 21 (LTS) to 23 (non-LTS) when the image tag is prefixed with the private registry.

sample Dockerfile snippet:

```
FROM eclipse-temurin:21-alpine@sha256:c63d8669d87e16bcee66c0379d1deedf844152da449ad48f2c8bd73a3705d36b
```

but will keep non-prefixed images on LTS as expected. Do I need to configure something specifically for Renovate to understand that private registry provided images should still follow the workaround?

**Comment by rarkins** [comment]
You'll need to customize https://github.com/renovatebot/renovate/blob/416ab8ec56076f0c52a11ad078fd17b1f11ebe29/lib/config/presets/internal/workarounds.ts#L131-L153 (copy/paste/update into your own config)

**Reply by cody-reibsome** [reply]
sorry - just trying to make sure I understand: do I need to mimic this and add a `registryUrls` to the `packageRules`? or is there another change I'm missing?

**Reply by rarkins** [reply, marked as answer]
You need to copy and paste it and modify the package name list to suit your needs

**Reply by cody-reibsome** [reply]
OK, that worked. Thank you.

It seems odd that the private registry is not automatically detected and removed from the package name - this means I'll need to go in and edit the list of allowed versions manually whenever a new LTS release occurs rather than relying on inheriting those values from the default configs. Do you know if this is being tracked as an issue somewhere?
